src/input/input_handler.cpp

```diff
@@ -0,0 +1,694 @@
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::RightY),
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::TriggerLeft),
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::TriggerRight),
+
+    // the "sorry, you gave an incorrect value, now we bind it to nothing" output
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::AxisMax),
+};
+
+// We had to go through 3 files of indirection just to update a flag
+void ToggleMouseEnabled() {
+    mouse_enabled ^= true;
+}
+
+// parsing related functions
+u32 GetAxisInputId(AxisMapping a) {
+    // LOG_INFO(Input, "Parsing an axis...");
+    if (a.axis == Axis::AxisMax || a.value != 0) {
+        LOG_ERROR(Input, "Invalid axis given!");
+        return 0;
+    }
+    u32 value = (u32)a.axis + 0x80000000;
+    return value;
+}
+
+u32 GetOrbisToSdlButtonKeycode(OrbisPadButtonDataOffset cbutton) {
+    switch (cbutton) {
+    case OrbisPadButtonDataOffset::Circle:
+        return SDL_GAMEPAD_BUTTON_EAST;
+    case OrbisPadButtonDataOffset::Triangle:
+        return SDL_GAMEPAD_BUTTON_NORTH;
+    case OrbisPadButtonDataOffset::Square:
+        return SDL_GAMEPAD_BUTTON_WEST;
+    case OrbisPadButtonDataOffset::Cross:
+        return SDL_GAMEPAD_BUTTON_SOUTH;
+    case OrbisPadButtonDataOffset::L1:
+        return SDL_GAMEPAD_BUTTON_LEFT_SHOULDER;
+    case OrbisPadButtonDataOffset::R1:
+        return SDL_GAMEPAD_BUTTON_RIGHT_SHOULDER;
+    case OrbisPadButtonDataOffset::L3:
+        return SDL_GAMEPAD_BUTTON_LEFT_STICK;
+    case OrbisPadButtonDataOffset::R3:
+        return SDL_GAMEPAD_BUTTON_RIGHT_STICK;
+    case OrbisPadButtonDataOffset::Up:
+        return SDL_GAMEPAD_BUTTON_DPAD_UP;
+    case OrbisPadButtonDataOffset::Down:
+        return SDL_GAMEPAD_BUTTON_DPAD_DOWN;
+    case OrbisPadButtonDataOffset::Left:
+        return SDL_GAMEPAD_BUTTON_DPAD_LEFT;
+    case OrbisPadButtonDataOffset::Right:
+        return SDL_GAMEPAD_BUTTON_DPAD_RIGHT;
+    case OrbisPadButtonDataOffset::Options:
+        return SDL_GAMEPAD_BUTTON_START;
+    case (OrbisPadButtonDataOffset)BACK_BUTTON:
+        return SDL_GAMEPAD_BUTTON_BACK;
+
+    default:
+        return ((u32)-1) - 0x10000000;
+    }
+}
+u32 GetControllerButtonInputId(u32 cbutton) {
+    if ((cbutton & ((u32)OrbisPadButtonDataOffset::TouchPad | LEFTJOYSTICK_HALFMODE |
+                    RIGHTJOYSTICK_HALFMODE)) != 0) {
+        return (u32)-1;
+    }
+    return GetOrbisToSdlButtonKeycode((OrbisPadButtonDataOffset)cbutton) + 0x10000000;
+}
+
+// syntax: 'name, name,name' or 'name,name' or 'name'
+InputBinding GetBindingFromString(std::string& line) {
+    u32 key1 = 0, key2 = 0, key3 = 0;
+
+    // Split the string by commas
+    std::vector<std::string> tokens;
+    std::stringstream ss(line);
+    std::string token;
+
+    while (std::getline(ss, token, ',')) {
+        tokens.push_back(token);
+    }
+
+    // Check and process tokens
+    for (const auto& t : tokens) {
```

Comment:
```suggestion
    for (const auto token : std::views::split(line, ","sv)) {
```
This will avoid allocations